### ISSUE TITLE
[torch.compile] Warning using BuiltinVariable.call_dict but not for {}

### ISSUE THREAD

**Issue by JonasGeiping** [pytorch/pytorch#96459]
### 🐛 Describe the bug

I would expect torch.compile to work equally well for dictionaries created using `dict` as for dictionaries created with `{}`, but using `dict` leads to the warning `incorrect arg count <bound method BuiltinVariable.call_dict of BuiltinVariable(dict)> got an unexpected keyword argument 'loss' and no constant handler`.

Is this expected behavior?

Here's a small reproduction:

```
class Model(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.embedding = torch.nn.Embedding(10, 8)

    def forward(self, input_ids):
        outputs = self.embedding(input_ids)
        loss = outputs.mean()
        return dict(loss=loss, outputs=outputs)  # warning
        # return {"loss": loss, "outputs": outputs} ok
 ```
test with
```
model = torch.compile(Model())
model(torch.randint(10, (4, 16)))
```

### Error logs

2.1.0.dev20230307
[2023-03-09 17:17:55,890] torch._dynamo.symbolic_convert: [INFO] Step 1: torchdynamo start tracing forward
[2023-03-09 17:17:55,895] torch._dynamo.variables.builtin: [WARNING] incorrect arg count <bound method BuiltinVariable.call_dict of BuiltinVariable(dict)> got an unexpected keyword argument 'loss' and no constant handler
[2023-03-09 17:17:55,895] torch._dynamo.output_graph: [INFO] Step 2: calling compiler function debug_wrapper
[2023-03-09 17:17:56,774] torch._inductor.compile_fx: [INFO] Step 3: torchinductor compiling FORWARDS graph 0
[2023-03-09 17:17:58,713] torch._inductor.compile_fx: [INFO] Step 3: torchinductor done compiling FORWARDS graph 0
[2023-03-09 17:17:58,713] torch._dynamo.output_graph: [INFO] Step 2: done compiler function debug_wrapper

### Versions

PyTorch version: 2.1.0.dev20230307
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @davidberard98

**Comment by ezyang** [comment]
We just forgot to implement the kwarg constructor. Should be simple enough. Happy to advise if someone wants to try.

**Comment by still-learning-ev** [comment]
i would like to be work on this issue.

**Comment by weshaan** [comment]
@ezyang hey, I was following various good first issues and made my way till here. I can try to help, pls explain me the steps with clarity and detail as I am a beginner, so I need slight assistance to understand things....I am looking forward to learn and help :)

**Comment by ezyang** [comment]
You need to find the code that handles dict constructor. It's probably in torch/_dynamo/variables/dict.py. There is some part that does argument handling with args/kwargs. You need to adjust it to handle the kwargs case.

**Comment by ezyang** [comment]
Don't forget to add tests.

**Comment by weshaan** [comment]
can you pls elaborate "add tests", exactly what I need to do and how I can do it effectively?

**Comment by weshaan** [comment]
Hey, I was reading the code and was trying to understand it, here are few lines that I think make sense related to the problem. Pls let me know your feedback... 
Thank you.

https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/dicts.py#L18-L25
https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/dicts.py#L76
https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/user_defined.py#L96-L98

**Comment by ezyang** [comment]
Put one in test/dynamo/test_repros.py

The dict constructor is actually probably in torch/_dynamo/variables/builtin.py specifically `call_dict`

**Comment by weshaan** [comment]
https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/builtin.py#L801-L804
https://github.com/pytorch/pytorch/blob/master/torch/_dynamo/variables/builtin.py#L748-L774

**Comment by weshaan** [comment]
I need help. I have been trying to read the code but i don't understand it, how will I fix it.... i did visit various sources to learn understanding open source code better but it is not helping me much here

**Comment by weshaan** [comment]
Hey, so I tried learning about this I have understood is that I need to do implement loss.kwarg() in a model to backpropagate the error. Pls let me know if i am going in the correct direction... Thank you.
